### PR TITLE
PLANET-6690 Remove Button block overrides from BlockFilters

### PR DESCRIPTION
**Pull request by mleray**
### Description

See [PLANET-6690](https://jira.greenpeace.org/browse/PLANET-6690)
This is only for the Button block, other block overrides will be handled in another PR. This was also an opportunity to remove the `Donate` style since we no longer want to have it available on our sites (see [PLANET-6698](https://jira.greenpeace.org/browse/PLANET-6698)).

**Related PR:** https://github.com/greenpeace/planet4-master-theme/pull/1662

### Testing

You can test this on your local or on the [titan test instance](https://www-dev.greenpeace.org/test-titan/), by adding a Button block in the backend. It should by default have the secondary style, and the only other available style should be `Primary` (`Donate` has been removed from the style picker, all the corresponding classes and styles will be removed later as part of [PLANET-6698](https://jira.greenpeace.org/browse/PLANET-6698)). Please also make sure the buttons are still correctly styled in the frontend too.

### CODE REVIEW

File: admin/js/editor.js [hunk 1]
```diff
@@ -12,4 +13,23 @@ wp.domReady(() => {
...
+  // Add our custom Button block styles
+  const styles = [
+    {
+      name: 'secondary',
+      label: __('Secondary', 'planet4-blocks-backend'),
+      isDefault: true
+    },
+    {
+      name: 'cta',
+      label: __('Primary', 'planet4-blocks-backend')
```
Comment (mleray) [inline admin/js/editor.js:30]: Mag agreed it made more sense to rename this style to `Primary` (it used to be `CTA`)

**Comment by planet-4** [comment]
### Test instance is ready :rocket:

:new_moon: [titan](https://www-dev.greenpeace.org/test-titan) | [admin](https://www-dev.greenpeace.org/test-titan/wp-admin/) | [blocks report](https://www-dev.greenpeace.org/test-titan/wp-admin/?page=plugin_blocks_report) | [CircleCI](https://app.circleci.com/pipelines/github/greenpeace/planet4-test-titan) | [composer-local.json](https://github.com/greenpeace/planet4-test-titan/blob/main/composer-local.json)

:watch: 2022.04.06 07:25:37

**Review by Inwerpsel — APPROVED** [review]
Looks good, nice to have the dead code removed too :100: 

This needs to be merged after https://github.com/greenpeace/planet4-master-theme/pull/1662, right?

**Comment by mleray** [comment]
> This needs to be merged after [greenpeace/planet4-master-theme#1662](https://github.com/greenpeace/planet4-master-theme/pull/1662), right?

Yep, or at least they need to be merged together!

**Pull request merged**